packages/types/src/wss.ts

```diff
@@ -179,3 +179,9 @@ export type ServerEventsMap = {
   [ServerEvent.ServerStats]: EventCallback<Server.Stats>;
   [ServerEvent.UserTyping]: EventCallback<{ roomId: number; userId: number }>;
 };
+
+// Generic types to extract the payload of the events
+export type ServerEventPayload<T extends keyof ServerEventsMap> =
+  ServerEventsMap[T] extends EventCallback<infer P> ? P : never;
+export type ClientEventPayload<T extends keyof ClientEventsMap> =
+  ClientEventsMap[T] extends EventCallback<infer P> ? P : never;
```

Comment:
The way you did it is much better what I initially suggested. I believe we take this a step further and compine these two types into one. 

Here is the code to do this. I tested it. The new type should accept any event type and will return its payload. 

> [!NOTE]
>  `EventCallbackPayload` is a private utility type. 

```ts
type EventCallbackPayload<T> = T extends EventCallback<infer P> ? P : never;

export type EventPayload<
  T extends keyof ClientEventsMap | keyof ServerEventsMap,
> = T extends keyof ClientEventsMap
  ? EventCallbackPayload<ClientEventsMap[T]>
  : T extends keyof ServerEventsMap
    ? EventCallbackPayload<ServerEventsMap[T]>
    : never;
```